### PR TITLE
add cf-python

### PR DESCRIPTION
**Pull request by ocefpaf**
https://pypi.org/project/cf-python/

Ping @Bru-od

### CODE REVIEW

**Comment by ocefpaf** [comment]
Needs https://github.com/conda-forge/cfunits-feedstock/pull/17

File: recipes/cf-python/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  sha256: d7622d60357811599352a31cce9d8055ed82fc3710db498ee2d376d52726d50a
+
+build:
+  number: 0
+  skip: True  # [win]
+  script: {{ PYTHON }} -m pip install . -vv
+
+requirements:
+  host:
+    - python >=3
```
Comment (marcelotrevisani) [inline recipes/cf-python/meta.yaml:20]:
```suggestion
    - python
```

File: recipes/cf-python/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    - pip
+  run:
+    - python >=3
```
Comment (marcelotrevisani) [inline recipes/cf-python/meta.yaml:23]:
```suggestion
    - python
```

File: recipes/cf-python/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: d7622d60357811599352a31cce9d8055ed82fc3710db498ee2d376d52726d50a
+
+build:
+  number: 0
+  skip: True  # [win]
```
Comment (marcelotrevisani) [inline recipes/cf-python/meta.yaml:15]:
```suggestion
  skip: True  # [win or py2k]
```

**Comment by marcelotrevisani** [comment]
uunnn... this recipe is an unusual one, it is not using the build extensions on setup.py, it is invoking make directly using a subprocess
I think we will need to add the compilers and make as well

File: recipes/cf-python/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+  script: {{ PYTHON }} -m pip install . -vv
+
+requirements:
```
Comment (marcelotrevisani) [inline recipes/cf-python/meta.yaml:18]:
```suggestion
requirements:
  build:
    - {{ compiler('c') }}
    - make
```

Comment (ocefpaf) [reply recipes/cf-python/meta.yaml:18]: They do import `Extension` but I believe that was a mistake b/c it is not used anywhere and I could not find any compiled extension in the source code. I'll check again though.

The problem here seems related to `cfunits`.

Comment (ocefpaf) [reply recipes/cf-python/meta.yaml:18]: Oh wait. I missed your comment above. Let me check it again.

**Comment by ocefpaf** [comment]
> uunnn... this recipe is an unusual one, it is not using the build extensions on setup.py, it is invoking make directly using a subprocess
> I think we will need to add the compilers and make as well

Yeah, this package is very odd. Sadly it is a nice thing and people who work with climate science on that institution needs it.

**Comment by ocefpaf** [comment]
@marcelotrevisani since you already reviewed it, can you take a second look and merge if it is OK?

**Pull request merged**